app/models/comment.rb

```diff
@@ -2,7 +2,7 @@ class Comment < ApplicationRecord
   belongs_to :user
   belongs_to :post
 
-  after_save :update_comment_counter
+  after_save :update_comments_counter
```

Comment:
Nice job adding validations and using the rails active call back as suggested by the previous reviewer😉. I am still having issues creating a comment because you used `author_id` in creating a post and requested `user_id` when generating a comment, instead of `author_id`.

<img width="634" alt="Screenshot 2024-01-10 at 08 26 31" src="https://github.com/hamidazim321/blog_app/assets/64643995/3dc082c4-23bc-47cb-9a70-4ddccd4c2062">

To resolve this, kindly stay consistent with `user_id` or `author_id` 

for instance, to change user_id in the comments table to author_id, you can implement it this way😉.

1. Create a migration file to update a column name by running. `rails generate migration RenameOldColumnNameToNewColumnName`
2. A migration file will be created, you can then edit it to look like this.
```
class RenameOldColumnNameToNewColumnName < ActiveRecord::Migration[7.1]
  def change
    rename_column :comments, :user_id, :author_id
  end
end

```
3. After this, run `rails db:migrate`😉.


If you do this for comments, you'll also have to do it for the likes table to ensure consistency👍🏼. Or you might just change `author_id` in the post table to `user_id`. whichever works for you👍🏼. 

But I'd recommend you make use of the first method as I have shown. change for the likes and comments as this worked for me.

Reply:
@Ibizugbe , I'm just following the ER Diagram given by Microverse and following the rule to name all fields same as the ones given in the ER Diagram. As you can see, in the ER Diagram given its author_id for post and user_id for comment.
![s](https://github.com/hamidazim321/blog_app/assets/129533107/e7305526-e046-4293-bfcd-95f99264a9a0)

Reply:
As you can see here in the project requirements @Ibizugbe 
![p](https://github.com/hamidazim321/blog_app/assets/129533107/553c2cb6-b856-4270-b104-9fcbb12f59bd)

Reply:
Alright, that's fine. thanks for pointing that out. I think you're right. I didn't take a look at this.  

<img width="750" alt="image" src="https://github.com/hamidazim321/blog_app/assets/64643995/9dc8ad67-6c7b-4962-bce9-448e5e216652">
This is the error I got when trying to create a comment and I fixed it that way, without taking a look at the ERD. You may disregard the method I have given you since it doesn't match the requirements👍🏼.

Reply:
Hello @hamidazim321 , Thanks for pointing this out👍🏼. I didn't take a look at the ERD diagram. you are right, your database matches the ERD diagram.  I set up the project again and tried creating a comment and it worked fine👍🏼. I think the issue was from me using `author_id` instead of `user_id`. Your project should have been approved and I am sorry for the inconvenience.  You can proceed to the other milestone without requesting a review. your project will be approved soon👍🏼.

<img width="749" alt="image" src="https://github.com/hamidazim321/blog_app/assets/64643995/15fae6e9-0686-4fa3-83c9-4355e0769c35">